### DISCUSSION TITLE
Tertiary colour for the Hybrid Coaster

### DISCUSSION THREAD

**Discussion by karst** [OpenRCT2/OpenRCT2#17393] [Feature requests]
![](https://pbs.twimg.com/media/FUp1T-yXsAEIcZP?format=jpg&name=4096x4096)
_Image by [@gavinwaple](https://twitter.com/gavinwaple)_

A week or two back I had a talk with @X123M3-256 on discord about having a tertiary colour on the hybrid coaster. I do think it's an interesting idea, which is why I am creating a discussion here.

@duncanspumpkin (I believe) mentioned that a singular sprite doesn't allow for three colours, only for two, or it'd get finicky. Currently the track and part of the supports are rendered onto the same sprite, which is why the tertiary colour isn't available currently.

To make this work the hybrid coaster's drawing code would have to be redone and have the track rendered separately from the supports. It might be cleaner that way, but I have no clue how much work that is. If it's worth it for X to put the effort into it and if he wants to do it in the first place. 

I personally think it'd be dope to have the ability to colour the two sides of tracks separately. We have several IRL RMC hybrid coasters that have this feature, like: "ArieForce One" and "The Joker".

All secondary colours (as the supports are usually tertiary, but only the first and third are in use currently) would be changed to the first colour when importing. 

I don't know what the current progress is on the new elements (Zero-G rolls, potentially large loop, and a few others) but it might be worth having the supports and track separated at around the same time those are added, if X would like to go through with this feature.

Anyways, I'd like to know the opinion of the devs and X if this is worth implementing.

**Comment by X123M3-256** [comment]
If I considered this worth implementing I would have done it the first time round. If you want to try implementing it I can send you a set of sprites with the supports and track separated or a set with a tertiary remap, but there's a lot of track paint code that would have to be modified and I don't intend on doing that.

**Reply by karst** [reply]
Hmm might be worth posting the set here. I'd love to experiment with it myself.
I'm getting quite familiar with the code of the track pieces. And if I can't figure it out myself perhaps someone else could pick it up or collaboratively with someone.

I think for it to work the track and supports need to be rendered separately from eachother with the two remappable colours on each rail (one yellow, one pink, i thought?) I don't quite remember what colour links to what anymore.

I do think it might be worth waiting till you have finalized the models for the new track elements though. That's a higher priority imo

**Comment by karst** [comment]
I am wondering how difficult it'd be to add a tertiary colour to the colour code for new rides to use if they want? That way the current sprites would only have to be replaced with one half rendered in another colour. Lets say light blue? Then no drawing code needs to be changed on the rides themselves and we could perhaps even see rides in the future featuring four colours if we really wanted?
